mapping_parenting_tech/analysis/prototyping/play_store/get_play_store_apps.py

```diff
@@ -449,266 +627,255 @@ def _process_review_grab(
-    """ """
+def load_all_app_reviews() -> pd.DataFrame():
+    """
+    Loads all app reviews into a Pandas dataframe.
 
-    review_list = list()
-    with open(folder / filename) as csv_reader:
-        reviews = csv.DictReader(csv_reader)
-        for review in reviews:
-            review_list.append(review)
+    This may take some time for 1,000s of apps (e.g., ~2.5minutes for reviews of 3,500 apps)
 
-    return review_list
+    Returns:
+        Pandas dataframe containing all reviews for all apps - note, this can be quite large!
+    """
 
+    all_reviews = pd.DataFrame()
+    review_files = REVIEWS_DIR.iterdir()
+    reviews_df_list = list()
+
+    logging.info("Loading files")
+    for file in tqdm(review_files):
+        reviews_df = pd.read_csv(file, header=0, index_col=None)
+        reviews_df_list.append(reviews_df)
+
+    logging.info("Concatenating data")
+    all_reviews = pd.concat(reviews_df_list, axis=0, ignore_index=True)
+
+    return all_reviews
 
-# %% [markdown]
-# ## Putting it together
-# ### Retrieving and using app ids via a set of downloaded web pages
 
 # %%
-def retrieve_app_set(seed: str, base_name: str, **kwargs):
+def list_update_all_app_reviews() -> set():
     """
-    Carries out the steps to retrieve the details for a given set of apps and saves the relevant output files.
+    Returns a set of apps whose ids are saved but not their reviews, plus apps whose review fetch is incomplete.
+
+    This function takes no arguments.
 
+    Returns a set of app ids, which can be passed straight to `save_playstore_app_list_reviews`
     """
 
-    file_names = {
-        "ids": base_name + "_ids.csv",
-        "details": base_name + "_details.json",
-        "reviews": base_name + "_reviews.csv",
-    }
+    t_df = load_all_app_reviews()
+    reviewed_apps = set(t_df.appId.to_list())
+    del t_df
 
-    options = {"ids": True, "details": True, "reviews": True}
-    options.update(kwargs)
+    all_apps = load_all_app_ids()
 
-    app_ids = parse_folder_for_app_ids(seed)
+    apps_to_get = [x for x in all_apps if x not in reviewed_apps]
 
-    if options["ids"]:
-        save_app_ids(app_ids, seed, file_names["ids"])
+    log_details = _init_log_info()
+    apps_to_get.extend(k for k, v in log_details.items() if v["completed"] == False)
 
-    if options["details"]:
-        save_app_details(get_playstore_app_details(app_ids), file_names["details"])
+    return set(apps_to_get)
 
-    if options["reviews"]:
-        save_playstore_app_list_reviews(app_ids, file_names["reviews"])
 
+# %% [markdown]
+# ## Putting it together
+# ### Retrieving and using app ids via a set of downloaded web pages
 
 # %%
-app_sets = {
-    "education_apps": "play_store/education_apps",
-    "parenting_apps": "play_store/parenting_apps",
-}
-
-for apps, folder in app_sets.items():
-    retrieve_app_set(folder, apps, ids=False, details=False)
+app_ids = parse_folder_for_app_ids("play_store/kids_under_five")
+update_all_app_id_list(app_ids)
```

Comment:
I tried to test this using a new data folder (so that I'm not overwriting any of your fetched data files) and this threw me an error, because I didn't have the `all_app_ids.csv'` file. Perhaps it can be created a new if the file doesn't exist?

Reply:
Have incorporated per suggestion - but into 'play_store_utils.py' in `utils`